crates/iroha/tests/integration/multisig.rs

```diff
@@ -77,76 +218,275 @@ fn mutlisig() -> Result<()> {
+    test_client.submit_blocking(Log::new(Level::DEBUG, "Just ticking time".to_string()))?;
+
+    // Check that the multisig transaction has executed
+    test_client
+        .query_single(FindAccountMetadata::new(msa_012345.clone(), key.clone()))
+        .expect("key-value should be set with enough approvals");
+
+    Ok(())
+}
+
+#[test]
+fn persistent_domain_level_authority() -> Result<()> {
+    let (_rt, _peer, test_client) = <PeerBuilder>::new().with_port(11_415).start_with_runtime();
+    wait_for_genesis_committed(&vec![test_client.clone()], 0);
+
+    let wonderland: DomainId = "wonderland".parse().unwrap();
+
+    let ms_accounts_registry_id = multisig_accounts_registry_of(&wonderland);
+
+    // Domain owner changes from Alice to Bob
+    test_client.submit_blocking(Transfer::domain(
+        ALICE_ID.clone(),
+        wonderland,
+        BOB_ID.clone(),
+    ))?;
+
+    // One block gap to follow the domain owner change
+    test_client.submit_blocking(Log::new(Level::DEBUG, "Just ticking time".to_string()))?;
+
+    // Bob is the authority of the wonderland multisig accounts registry
+    let ms_accounts_registry = test_client
         .query(FindTriggers::new())
-        .filter_with(|trigger| trigger.id.eq(multisig_trigger_id.clone()))
+        .filter_with(|trigger| trigger.id.eq(ms_accounts_registry_id.clone()))
         .execute_single()
-        .expect("multisig trigger should be created after the call to register multisig trigger");
+        .expect("multisig accounts registry should survive before and after a domain owner change");
 
-    assert_eq!(trigger.id(), &multisig_trigger_id);
+    assert!(*ms_accounts_registry.action().authority() == BOB_ID.clone());
 
-    let asset_definition_id = "asset_definition_controlled_by_multisig#wonderland"
-        .parse::<AssetDefinitionId>()
-        .unwrap();
-    let isi =
-        vec![
-            Register::asset_definition(AssetDefinition::numeric(asset_definition_id.clone()))
-                .into(),
-        ];
-    let isi_hash = HashOf::new(&isi);
-
-    let mut signatories_iter = signatories.into_iter();
-
-    if let Some((signatory, key_pair)) = signatories_iter.next() {
-        let args = MultisigArgs::Instructions(isi);
-        let call_trigger = ExecuteTrigger::new(multisig_trigger_id.clone()).with_args(&args);
-        test_client.submit_transaction_blocking(
-            &TransactionBuilder::new(test_client.chain.clone(), signatory)
-                .with_instructions([call_trigger])
-                .sign(key_pair.private_key()),
-        )?;
+    Ok(())
+}
+
+#[test]
+fn reserved_names() {
+    let (_rt, _peer, test_client) = <PeerBuilder>::new().with_port(11_420).start_with_runtime();
+    wait_for_genesis_committed(&vec![test_client.clone()], 0);
+
+    let account_in_another_domain = gen_account_in("garden_of_live_flowers").0;
+
+    {
+        let reserved_prefix = "multisig_accounts_";
```

Comment:
I would make the prefix more "special" like `__MULTISIG_ACCOUNT__`

Reply:
I think this suggestion can be connected to the scoping discussion #5160